pipeline/pipeline/pipeline.py

```diff
@@ -82,19 +135,107 @@ def _new_resource_group(self, source, mappings):
         return rg
 
     def read_input(self, path, extension=None):
+        """
+        Create a new input resource file object representing a single file.
+
+        Examples
+        --------
+
+        Read the file `hello.txt`:
+
+        >>> p = Pipeline()
+        >>> input = p.read_input('hello.txt')
+        >>> t = p.new_task()
+        >>> t.command(f"cat {input}")
+        >>> p.run()
+
+        Parameters
+        ----------
+        path: :obj:`str`
+            File path to read.
+        extension: :obj:`str`, optional
+            File extension to use.
+
+        Returns
+        -------
+        :class:`.InputResourceFile`
+        """
+
         irf = self._new_input_resource_file(path)
         if extension is not None:
             irf.add_extension(extension)
         return irf
 
     def read_input_group(self, **kwargs):
+        """
+        Create a new resource group representing a mapping of identifier to
+        input resource files.
+
+        Examples
+        --------
+
+        Read a binary PLINK file:
+
+        >>> p = Pipeline()
+        >>> bfile = p.read_input_group(bed="data/example.bed",
+        ...                            bim="data/example.bim",
+        ...                            fam="data/example.fam")
+        >>> t = p.new_task()
+        >>> t.command(f"plink --bfile {bfile} --geno --out {t.geno}")
+        >>> t.command(f"wc -l {bfile.fam}")
+        >>> t.command(f"wc -l {bfile.bim}")
+        >>> p.run()
```

Comment:
Adding feedback from @konradjk in person so I have a record of changes to make:

- Make clear that the file extensions come from the identifiers
- Give example of plink and fasta.idx
- Make clear if don't care what file extensions are, can just choose any arbitrary identifier -- used only when referencing a particular file

Also, interface change suggestion:
```
p.read_input_group(fasta="example.fasta",
                                 index=("example.fasta.idx", "{root}.fa.idx"))
```

Also, might be good to make interface for `declare_resource_group` and `read_input_group` have the same name and similar interface with kwargs.